### ISSUE TITLE
Usability Showstoppers

### ISSUE THREAD

**Issue by p-a-s-c-a-l** [clarity-h2020/docker-drupal#141]
Here we collect usability-related showstopper issues that have to be resolved before we can release the [public BETA release](https://github.com/clarity-h2020/csis/issues/145) of CSIS and perform the [internal acceptance tests](https://github.com/clarity-h2020/csis-technical-validation/issues/1).

**Comment by p-a-s-c-a-l** [comment]
We have to remove or hide all non-working and dysfunctional study types:

![grafik](https://user-images.githubusercontent.com/1788226/75146278-0d693480-56fb-11ea-8a7a-550b578a0f18.png)

In fact, the only study types that will soon become usable by external users are are **Basic Screening: General** and **Automated Screening: Urban Infrastructure**. Expert Studies are currently for our DCs only. So we have find a way, to hide those studies from the list. 

@fgeyer16 Perhaps this has already been revolved in https://github.com/clarity-h2020/docker-drupal/issues/115 ? ATM, I'm not able to test it because I cannot log in with another user due to https://github.com/clarity-h2020/docker-drupal/issues/140

**Comment by patrickkaleta** [comment]
> In fact, the only study types that will soon become usable by external users are are **Basic Screening: General** and **Automated Screening: Urban Infrastructure**. Expert Studies are currently for our DCs only. So we have find a way, to hide those studies from the list.
> 
> @fgeyer16 Perhaps this has already been revolved in #115 ? ATM, I'm not able to test it because I cannot log in with another user due to #140

Yes, this has already be resolved. A "normal" logged in user can see only these study types:
![select-studytype](https://user-images.githubusercontent.com/42167036/75147086-d1cf6a00-56fc-11ea-9225-3fdf9b467340.png)


I will deactivate the "Advanced Screening: Transport Infrastructure" until the [integration of the TM](https://github.com/clarity-h2020/csis-helpers-module/issues/19) is done.

BTW: #140 is not happening to me anymore (I did have problems with the CAS earlier this morning).

**Comment by p-a-s-c-a-l** [comment]
Add Study and Edit Study Forms are different but probably they shouldn't:

![grafik](https://user-images.githubusercontent.com/1788226/75158188-e409d200-5715-11ea-8cd2-b0b078ca733c.png)

Most of the information in the Edit Study Form are **required** for successfully performing or assessing a study (country, city, scenarios). But those important settings are "hidden" behind the "edit" Button of the study context page. So users will never know that they **must** enter this information and will get stuck e.g. here:

![grafik](https://user-images.githubusercontent.com/1788226/75161324-dfe0b300-571b-11ea-86e3-0fe2ce0eb005.png)

So IMHO this is a serious usability issue that could render the system impossible to use for anybody not involved in its implementation. A straightforward solution could be to use the same form for *add* and *edit*. But this does not solve the more general problem, that there isn't any workflow guidance as we intended to provide with help of the [Step-by-Step Wizard](https://github.com/clarity-h2020/docker-drupal/issues/72). I think we have to put more emphasis on this now.

**Comment by p-a-s-c-a-l** [comment]
Coming back to country selection, there are two usability issues:

![grafik](https://user-images.githubusercontent.com/1788226/75223947-d4d36480-57a7-11ea-9c84-5a9c95729d64.png)

The first, more serious one is related to the [recalculation of HC-LE input layers](https://github.com/clarity-h2020/emikat/issues/32): The city selection box must contain cities only, that are guaranteed to be available (recalculated) for screening. So we have to periodically re-generate the cities taxonomy as @therter described [here](https://github.com/clarity-h2020/docker-drupal/issues/68#issuecomment-523336896). I wonder, if there is a possibility to a) automate or b) simplify this process? If we could unpublish or hide some taxonomy terms, then it would still be a manual task to update the list of cities as new cities are becoming available, but it's still more efficient than doing the complete import process again and again.

The second issue is not that critical (showstopper) but it could still become frustrating for users:  There are many *countries* without supported cities, so they shouldn't be shown at all in the countries selection box for **Automated Screening: Urban Infrastructure** studies.

**Comment by p-a-s-c-a-l** [comment]
> users will never know that they **must** enter this information and will get stuck

O.K. I completely forgot that we've got those videos:

- [The CLARITY methodology explained](https://www.youtube.com/watch?v=_MvO_MqTPA0)
- [CLARITY CSIS: Create a profile](https://www.youtube.com/watch?v=PQOr0diQrPI)
- [CLARITY CSIS: Create a study](https://www.youtube.com/watch?v=AyfIRvPuZOM)

While this doesn't solve the most obvious usability issues, users can get at least an impression on how to use the system without getting stuck somewhere in between. 

So we can put [this issue](https://github.com/clarity-h2020/docker-drupal/issues/141#issuecomment-590354829) on hold. But we have to **link or embed those videos very prominently**,  e.g. before creating a new user profile or a new study.

**Comment by patrickkaleta** [comment]
> I wonder, if there is a possibility to a) automate or b) simplify this process? If we could unpublish or hide some taxonomy terms, then it would still be a manual task to update the list of cities as new cities are becoming available, but it's still more efficient than doing the complete import process again and again.

It is possible to manage the City taxonomy by hand. [Here](https://csis.myclimateservice.eu/maintenance/cities_regions) you can check for all cities the status and either publish or unpublish them. But with each re-generation using the process implemented by @therter this manual configuration of the taxonomy will be lost.

**Comment by p-a-s-c-a-l** [comment]
Users are able to create invalid study scenarios, e.g. by selecting a RCP even if the time period is historical:

![grafik](https://user-images.githubusercontent.com/1788226/75239860-b2e7db00-57c3-11ea-9c9f-6cd4b16e06f5.png)

Is there any possibility to implement a quick and simple validation for that? Otherwise we have to find another solution. e.g. at least add a static warning message. Or maybe use another list with predefined and valid TP/RCP combinations (if it's possible to set two different values from one selection list)?

**Comment by p-a-s-c-a-l** [comment]
Users **must not** be able to select a data package that will not work with the current study type:

![grafik](https://user-images.githubusercontent.com/1788226/75241177-eb88b400-57c5-11ea-8734-38a33a30b1a0.png)

ATM, just the **European Wide Data Package** is usable with the **Automated Screening: Urban Infrastructure** study. 

Note: This usability issue is already resolved in in https://github.com/clarity-h2020/csis/issues/91#issuecomment-590847973

**Comment by p-a-s-c-a-l** [comment]
Vulnerability ... This is still kind of a black box. No vulnerability-related resources available in the [European Data Package](https://csis.myclimateservice.eu/node/682), thus Data Tab for VA step is empty. Moreover, the vulnerability functions in the table look just like examples, not like real vulnerability functions (*TODO: add some text*):

![grafik](https://user-images.githubusercontent.com/1788226/75330869-0e789e00-5882-11ea-8e89-4ad3bd0c0c4f.png)

Apparently, those vulnerability functions are [not (yet?) used](https://github.com/clarity-h2020/emikat/issues/16) at all for impact calculation in EMIKAT. Can anybody shed some light on this, maybe @humerh @DenoBeno @bernhardsk ?

If there aren't any *real* VA functions available, we should remove this step until the situation has changed.

**Comment by p-a-s-c-a-l** [comment]
Twins ....

We've invested quite some effort in the implementation of the twins functionality, but without ever having a clear concept for getting real content (Hazard Twins, Exposure Twins) into the system. I don't think that this has changed and that there plans for (automatically) importing some Hazard Twins and Exposure Twins from data sources in the near future (=ASAP) @Itsman-AT ?

Without data, this is just another "proof-of-concept" that "just works" but nothing that provides real added value for end users.  Although it's a pity, but if the situation doesn't improve it may be better to remove the twins from PROD CSIS.  We can keep it enabled in DEV CSIS just for demonstration purposes, and can enable it again in PROD when we have something to offer apart from some test and demo entries.

**Comment by p-a-s-c-a-l** [comment]
Not a real showstopper, but something that should be fixed:

![grafik](https://user-images.githubusercontent.com/1788226/75677300-68ff6900-5c8b-11ea-930d-b37d05939877.png)

Scenario Selection is not needed here, in fact, it has no influence on the list of adaptation options, so it should be hidden.

**Comment by therter** [comment]
Regarding  [this ](https://github.com/clarity-h2020/docker-drupal/issues/141#issuecomment-590727856) issue
> So we have to periodically re-generate the cities taxonomy as @therter described [here](https://github.com/clarity-h2020/docker-drupal/issues/68#issuecomment-523336896). I wonder, if there is a possibility to a) automate or b) simplify this process?


I have created the script republish.sh. This script unpublishes all cities, which are not contained in the [city shape file](http://services.clarity-h2020.eu:8080/geoserver/clarity/wfs?request=GetFeature&typeNames=clarity:city&outputFormat=SHAPE-ZIP) or the field heat_wave is false.

To run the script, the following postgres function should exists in the db:

```
CREATE OR REPLACE FUNCTION public.publish_cities(
	)
    RETURNS boolean
    LANGUAGE 'plpgsql'

    COST 100
    VOLATILE 
AS $BODY$

declare
begin
	--publish/unpublish the cities. The data is contained in the table raw.cities
	update taxonomy_term_field_data r set status = case when exists(select 1 from raw.cities where name ilike r.name and heat_wave is not null) then 1 else 0 end where tid in (select tid from taxonomy_term_data where vid = 'cities_regions');
	update taxonomy_term_field_revision r set status = case when exists(select 1 from raw.cities where name ilike r.name and heat_wave is not null) then 1 else 0 end where tid in (select tid from taxonomy_term_data where vid = 'cities_regions');
        --publish/unpublish countries
	update taxonomy_term_field_data re set status = case when exists(
		select 1 from taxonomy_term_field_revision r  join taxonomy_term__field_country country on (country.field_country_target_id = r.tid) join taxonomy_term_field_data city on (country.entity_id = city.tid) where city.status = 1 and r.tid = re.tid
	) then 1 else 0 end where tid in (select tid from taxonomy_term_data where vid = 'countries');
	update taxonomy_term_field_revision re set status = case when exists(
		select 1 from taxonomy_term_field_revision r  join taxonomy_term__field_country country on (country.field_country_target_id = r.tid) join taxonomy_term_field_data city on (country.entity_id = city.tid) where city.status = 1 and r.tid = re.tid
	) then 1 else 0 end where tid in (select tid from taxonomy_term_data where vid = 'countries');
	return true;
end

$BODY$;
```


The republish.sh script can be found on the server csis.myclimateservice.eu in the directory home/thorsten . I have not executed it on dev server yet.

**Comment by p-a-s-c-a-l** [comment]
![grafik](https://user-images.githubusercontent.com/1788226/76227210-3fac8300-621f-11ea-945e-f9151fdb6af9.png)

This footer should be removed as it is no longer needed.

**Comment by p-a-s-c-a-l** [comment]
No a real showstopper but still annoying: We should disable the quick edit module (which [doesn't seem that easy](https://drupal.stackexchange.com/questions/192506/how-do-i-disable-the-quick-edit-module)). 

**Reason:** Regular Users don't see it (but the JS is executed in the background) and admins don't need it. This module seems to inject a lot of AJAX stuff into the pages which makes page loading not only slower but interferes also with [UI Integration Tests](https://github.com/clarity-h2020/csis-technical-validation/issues/13#issuecomment-596950447). This is a [known problem](https://www.drupal.org/project/drupal/issues/2938308), btw.

**Comment by p-a-s-c-a-l** [comment]
[Toggle edit](https://github.com/clarity-h2020/csis/issues/104#issuecomment-597057220) of Map Component can cause too many [EMIKAT re-calculations](https://github.com/clarity-h2020/emikat/issues/34). See also https://github.com/clarity-h2020/map-component/issues/87

**Comment by p-a-s-c-a-l** [comment]
Re(Calculation) is a serious issue. There is no clear indication when and why a re-calculation was triggered or not, e.g. it is not triggered when no data package is set, it is **immediately** triggered when the user [changes the box of the study area in the map](https://github.com/clarity-h2020/map-component/issues/87), etc. So [this feature](https://github.com/clarity-h2020/emikat/issues/34#issuecomment-595674938) has to be implemented before going live.

**Comment by patrickkaleta** [comment]
> Users are able to create invalid study scenarios, e.g. by selecting a RCP even if the time period is historical:
> 
> ![grafik](https://user-images.githubusercontent.com/1788226/75239860-b2e7db00-57c3-11ea-9c9f-6cd4b16e06f5.png)
> 
> Is there any possibility to implement a quick and simple validation for that? Otherwise we have to find another solution. e.g. at least add a static warning message. Or maybe use another list with predefined and valid TP/RCP combinations (if it's possible to set two different values from one selection list)?

For now, I've added a text under the emission scenario selection telling the users to select meaningful combinations. Other solutions would require changing the data model, which would also require changes in the CSIS helpers module, so that it could continue to provide the selected scenario in the `$studyInfo` object. 
If we find the time, we can adress this issue at some later point.

**Comment by patrickkaleta** [comment]
> ![grafik](https://user-images.githubusercontent.com/1788226/76227210-3fac8300-621f-11ea-945e-f9151fdb6af9.png)
> 
> This footer should be removed as it is no longer needed.

This footer is created in a [custom block](https://csis.myclimateservice.eu/block/3?destination=/admin/structure/block/block-content), which I've now disabled.

**Comment by patrickkaleta** [comment]
> No a real showstopper but still annoying: We should disable the quick edit module (which [doesn't seem that easy](https://drupal.stackexchange.com/questions/192506/how-do-i-disable-the-quick-edit-module)).

Quick edit module is **now uninstalled** (in the mentioned link, the person actually wanted to get rid of all edit icons, not just the quick-edit). 

The error message "_Uncaught DOMException: Failed to execute 'setItem'.._." is now gone.

**Comment by p-a-s-c-a-l** [comment]
Most of the usability showstoppers are resolved now, but one big issue remains [Twins](https://github.com/clarity-h2020/docker-drupal/issues/141#issuecomment-591344047). Apart form the lack of useful content, the Twins Map isn't working properly due missing [API key](https://github.com/clarity-h2020/docker-drupal/issues/146):

![grafik](https://user-images.githubusercontent.com/1788226/80345880-2bbdee80-886a-11ea-88c3-72a2e73200e9.png)

@Itsman-AT Any chance that this can be resolved? If not, we have to hide the twins tabs in the public beta.

**Comment by patrickkaleta** [comment]
> Most of the usability showstoppers are resolved now, but one big issue remains [Twins](https://github.com/clarity-h2020/docker-drupal/issues/141#issuecomment-591344047). Apart form the lack of useful content, the Twins Map isn't working properly due missing [API key](https://github.com/clarity-h2020/docker-drupal/issues/146):

We switched the Twins Maps to use Leaflet now instead, so no Google API key is required anymore.

Regarding the usefulness of the currently available Twins in the system, I don't know for sure how much added value they bring for the users. I don't think they are completely useless, so we might just keep them for now and await initial feedback from users.

**Comment by p-a-s-c-a-l** [comment]
Google Maps are also referenced in the Data package / resource views, e.g. here:
https://csis.myclimateservice.eu/study/35/view/data

At least for the data package view having a 2nd map which just shows the spatial coverage of the DP is somewhat redundant. So it might be better to remove at this GoogleMap instead of replacing it by leaflet.

**Comment by patrickkaleta** [comment]
You're right, there's no need to show the same info twice. I removed the Google map.
